lib/locreator.js

```diff
@@ -44,56 +48,63 @@ Locreator.prototype.addGenericMethods = function generics() {
   };
 };
 
-Locreator.prototype.addStarMethods = function locreator(locatorId, _locator, parentWebElement) {
+Locreator.prototype.addStarMethods = function addStarMethods(locatorId, locatorJSON, parentWebElement) {
+  log('add star methods for %s', locatorId);
   var locatorObject = {};
   var drivex = this.drivex;
-  var locator = this.normify(_locator);
+  var locreator = this;
+  var locator = function () {
+    return locreator.normify(locreator.locatex(locatorJSON));
+  };
```

Comment:
Just to provide some more clarity here. "Locale" (really specialization) would change on the fly in a data-driven scenario. Currently the plugin lifecycle is tied to the webdriver lifecycle and the dynamically generated methods create a closure around their particular locator. So, unless the dynamic methods themselves are recreated when “locale” changes (which would be a significant refactor) this would be the only way to reflect the change to the locator.

The proposal in https://github.com/paypal/nemo-view/issues/33 would provide an avenue for recreating the dynamic methods when specialization changes, if we deem that there is a significant impact to performance to generate the locator on the fly.